RecaptchaValidation/Services/RecaptchaService.cs

```diff
@@ -1,44 +1,41 @@
 ﻿using Microsoft.Extensions.Options;
 using RecaptchaValidation.Interfaces;
 using RecaptchaValidation.Models;
+using System;
+using System.IO;
+using System.Net.Http;
 using System.Runtime.Serialization.Json;
+using System.Threading.Tasks;
 
 namespace RecaptchaValidation.Services
 {
     public class RecaptchaService : IRecaptchaService
     {
         public HttpClient _httpClient { get; set; }
-        public IRecaptchaRequestMessage Request {get; set; }
-        public IRecaptchaResponseMessage Response { get; set; }
 
-        public RecaptchaService(HttpClient httpClient, IOptions<RecaptchaOptions> recaptchaOptions) {
+        public RecaptchaService(HttpClient httpClient) {
             _httpClient= httpClient;
         }
 
-        public void InitializeRequest(IRecaptchaRequestMessage request)
-        {
-            Request = request;
-        }
-
         public Task<byte[]> GetResponseContentAsByteArray(HttpResponseMessage responseMessage)
         {
             return responseMessage.Content.ReadAsByteArrayAsync();
         }
 
-        public async Task<IRecaptchaResponseMessage> Execute()
+        public async Task<RecaptchaResponseMessage> Execute(RecaptchaRequestMessage requestMessage)
         {
             try 
-            { 
-                string recaptchaVerificationUrl = Request.ToString();
+            {
+                RecaptchaResponseMessage Response;
+
+                string recaptchaVerificationUrl = RecaptchaRequestMessage.GetVerificationUrl(requestMessage);
 
-                HttpResponseMessage verificationResponse = await _httpClient.PostAsync(recaptchaVerificationUrl, null);
+                HttpResponseMessage verificationResponse = await _httpClient.PostAsync(recaptchaVerificationUrl, null).ConfigureAwait(false);
                 
                 verificationResponse.EnsureSuccessStatusCode();
 
                 byte[] responseData = await verificationResponse.Content.ReadAsByteArrayAsync();
```

Comment:
All awaits should have a `ConfigureAwait(false)` appended to them. This and the ones down below as well.